### PR TITLE
Create a list of build cluster service accounts we want to exist

### PR DESCRIPTION
**Pull request by fejta**
Also document how to use `experiment/workload-identity`

/assign @listx @Katharine

### CODE REVIEW

File: prow/cluster/build_serviceaccounts.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  annotations:
+    iam.gke.io/gcp-service-account: k8s-infra-gcr-promoter@k8s-gcr-audit-test-prod.iam.gserviceaccount.com
```
Comment (cjwagner) [inline prow/cluster/build_serviceaccounts.yaml]: nit: Do we want people to comment or annotate these with their intended uses in order to discourage permission aggregation and misuse? If so we should start setting a precedent for that and encourage it in the docs.

File: experiment/workload-identity/README.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  --role=roles/iam.workloadIdentityUser \
+  --member=serviceAccount:K8S_PROJECT.svc.id.goog[SOMEWHERE/SOMETHING] \
+  SOMEBODY@PROJECT.iam.gserviceaccount.com
+```
+
+This is what tells GCP that the `SOMEWHERE/SOMETHING` service account in
+`K8S_PROJECT` is authorized to act as
+`SOMEBODY@PROJECT.iam.gserviceaccount.com`.
+
+These are all describe in the how-to GKE doc link at the top.
```
Comment (listx) [inline experiment/workload-identity/README.md]: described

File: experiment/workload-identity/README.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+### Pods
+
+At this point any pod that:
+* Runs in a `K8S_PROJECT` GKE cluster
+* Inside the `SOMEWHERE` namespace
+* Using the `SOMETHING` service
+
+Will authenticate as `SOMEBODY@PROJECT.iam.gserviceaccount.com` to GCP. The
```
Comment (listx) [inline experiment/workload-identity/README.md]: s/Will/will

File: experiment/workload-identity/README.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+`bind-service-accounts.sh` script will verify this (see the how-to doc above for
+the manual command).
+
+Whenever you want a pod to authenticate this way, just configure the
+`serviceAccountName` on the `PodSpec`. See an example
+[pod](https://github.com/GoogleCloudPlatform/testgrid/blob/5c7bc80b18ccf00c773c34583628091890b401ab/cluster/summarizer_deployment.yaml#L22)
+and
+[prowjob](https://github.com/GoogleCloudPlatform/oss-test-infra/blob/9c466c14e4b4b5fbc8c837d0c61c779194e82d56/prow/prowjobs/GoogleCloudPlatform/oss-test-infra/gcp-oss-test-infra-config.yaml#L65):
+
+Here are minimal pod and prow job that use an image and args to print the
```
Comment (cjwagner) [inline experiment/workload-identity/README.md]: nit: `s/pod/deployment/` (although a podspec would be just as useful and more concise)

File: prow/cluster/build_serviceaccounts.yaml (hunk 1 above)
Comment (fejta) [reply prow/cluster/build_serviceaccounts.yaml]: Done

File: experiment/workload-identity/README.md (hunk 4 above)
Comment (fejta) [reply experiment/workload-identity/README.md]: Done

File: experiment/workload-identity/README.md (hunk 2 above)
Comment (fejta) [reply experiment/workload-identity/README.md]: Done

File: experiment/workload-identity/README.md (hunk 3 above)
Comment (fejta) [reply experiment/workload-identity/README.md]: Done

**Pull request merged**